pallets/merkle/src/lib.rs

```diff
@@ -264,33 +277,60 @@ impl<T: Config> Group<T::AccountId, T::BlockNumber, T::GroupId> for Module<T> {
 			Error::<T>::InvalidTreeDepth
 		);
 
+		// Setting the next group id
 		let group_id = Self::next_group_id();
 		NextGroupId::<T>::mutate(|id| *id += One::one());
 
-		let mtree = GroupTree::<T>::new(sender, is_manager_required, depth);
+		// Setting up the tree
+		let mtree = GroupTree::new::<T>(depth);
 		<Groups<T>>::insert(group_id, mtree);
+
+		// Setting up the manager
+		let manager = Manager::<T>::new(sender, is_manager_required);
+		<Managers<T>>::insert(group_id, manager);
 		Ok(group_id)
 	}
 
+	fn set_stopped(sender: T::AccountId, id: T::GroupId, stopped: bool) -> Result<(), dispatch::DispatchError> {
+		let manager_data = <Managers<T>>::get(id).ok_or(Error::<T>::ManagerDoesntExist).unwrap();
+		ensure!(sender == manager_data.account_id, Error::<T>::ManagerIsRequired);
+		Stopped::<T>::insert(id, stopped);
+		Ok(())
+	}
+
 	fn set_manager_required(
 		sender: T::AccountId,
 		id: T::GroupId,
 		manager_required: bool,
 	) -> Result<(), dispatch::DispatchError> {
-		let mut tree = <Groups<T>>::get(id).ok_or(Error::<T>::GroupDoesntExist).unwrap();
+		let mut manager_data = <Managers<T>>::get(id).ok_or(Error::<T>::ManagerDoesntExist).unwrap();
 		// Changing manager required should always require an extrinsic from the
 		// manager even if the group doesn't explicitly require managers for
 		// other calls.
-		ensure!(sender == tree.manager, Error::<T>::ManagerIsRequired);
-		tree.manager_required = manager_required;
+		ensure!(sender == manager_data.account_id, Error::<T>::ManagerIsRequired);
+		manager_data.required = manager_required;
+		<Managers<T>>::insert(id, manager_data);
+		Ok(())
+	}
+
+	fn set_manager(
+		sender: T::AccountId,
```

Comment:
Do we need this since we handle authentication in the pallet function itself?

Reply:
Since it seems we do a double authentication, it might be worthy to just keep the trait definition separate from the sender, leaving the pallet up to deciding how to change managers.